extensions/azure/assetindex-cosmos/src/main/java/org/eclipse/dataspaceconnector/assetindex/azure/CosmosAssetQueryBuilder.java

```diff
@@ -0,0 +1,54 @@
+package org.eclipse.dataspaceconnector.assetindex.azure;
+
+import org.eclipse.dataspaceconnector.assetindex.azure.model.AssetDocument;
+import org.eclipse.dataspaceconnector.cosmos.azure.CosmosDocument;
+import org.eclipse.dataspaceconnector.spi.EdcException;
+import org.eclipse.dataspaceconnector.spi.asset.AssetSelectorExpression;
+import org.eclipse.dataspaceconnector.spi.asset.Criterion;
+import org.eclipse.dataspaceconnector.spi.types.domain.asset.Asset;
+import org.jooq.Condition;
+import org.jooq.SQLDialect;
+import org.jooq.impl.DSL;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.jooq.impl.DSL.field;
+
+public class CosmosAssetQueryBuilder {
+
+    private static final String ASSET_FIELD = "wrappedInstance";
+    private static final String PROPERTIES_FIELD = "properties";
+    private static final String PATH_TO_PROPERTIES = String.join(".", AssetDocument.class.getSimpleName(), ASSET_FIELD, PROPERTIES_FIELD);
+
+    public CosmosAssetQueryBuilder() {
+        // pre-check fields of the AssetDocument class that will be used to build the queries
+        assertClassContainsField(CosmosDocument.class, ASSET_FIELD);
+        assertClassContainsField(Asset.class, PROPERTIES_FIELD);
+    }
+
+    public String from(AssetSelectorExpression expression) {
+        List<Condition> conditions = expression == AssetSelectorExpression.SELECT_ALL ? Collections.emptyList() : createConditions(expression);
+        return DSL.using(SQLDialect.MYSQL)
+                .select(field("*"))
+                .from(AssetDocument.class.getSimpleName())
+                .where(conditions)
+                .getSQL();
+    }
+
+    private List<Condition> createConditions(AssetSelectorExpression expression) {
+        return expression.getCriteria().stream().map(this::createCondition).collect(Collectors.toList());
+    }
+
+    private Condition createCondition(Criterion criterion) {
+        return field(PATH_TO_PROPERTIES + "." + criterion.getOperandLeft()).eq(field((String) criterion.getOperandRight()));
+    }
```

Comment:
Sorry, it's me again :) Maybe you can validate the criterion operator here. So that it throws an exception if the criterion requires a non-equal operation.